src/swiss_dict.jl

```diff
@@ -0,0 +1,559 @@
+             
+const _u8x16 = NTuple{16, VecElement{UInt8}}
+
+mutable struct SwissDict{K,V} <: AbstractDict{K,V}
+    slots::Vector{_u8x16}
+    keys::Vector{K}
+    vals::Vector{V}
+    nbfull::Int
+    count::Int
+    age::UInt
+    idxfloor::Int  # an index <= the indices of all used slots
+
+    function SwissDict{K,V}() where V where K
+        new(fill(_expand16(0x00),1), Vector{K}(undef, 16), Vector{V}(undef, 16), 0, 0, 0, 1)
+    end
+    function SwissDict{K,V}(d::SwissDict{K,V}) where V where K
+        new(copy(d.slots), copy(d.keys), copy(d.vals), d.nbfull, d.count, d.age,
+            d.idxfloor)
+    end
+    function SwissDict{K, V}(slots, keys, vals, nbfull, count, age, idxfloor) where {K, V}
+        new(slots, keys, vals, nbfull, count, age, idxfloor)
+    end
+end
+function SwissDict{K,V}(kv) where V where K
+    h = SwissDict{K,V}()
+    for (k,v) in kv
+        h[k] = v
+    end
+    return h
+end
+SwissDict{K,V}(p::Pair) where {K,V} = setindex!(SwissDict{K,V}(), p.second, p.first)
+function SwissDict{K,V}(ps::Pair...) where V where K
+    h = SwissDict{K,V}()
+    sizehint!(h, length(ps))
+    for p in ps
+        h[p.first] = p.second
+    end
+    return h
+end
+# Note the constructors of WeakKeyDict mirror these here, keep in sync.
+SwissDict() = SwissDict{Any,Any}()
+SwissDict(kv::Tuple{}) = SwissDict()
+copy(d::SwissDict) = SwissDict(d)
+empty(d::SwissDict, ::Type{K}, ::Type{V}) where {K, V} = SwissDict{K, V}()
+
+const AnyDict = SwissDict{Any,Any}
+
+SwissDict(ps::Pair{K,V}...) where {K,V} = SwissDict{K,V}(ps)
+SwissDict(ps::Pair...)                  = SwissDict(ps)
+
+function SwissDict(kv)
+    try
+        dict_with_eltype((K, V) -> SwissDict{K, V}, kv, eltype(kv))
+    catch
+        if !isiterable(typeof(kv)) || !all(x->isa(x,Union{Tuple,Pair}),kv)
+            throw(ArgumentError("SwissDict(kv): kv needs to be an iterator of tuples or pairs"))
+        else
+            rethrow()
+        end
+    end
+end
+
+function grow_to!(dest::AbstractDict{K, V}, itr) where V where K
+    y = iterate(itr)
+    y === nothing && return dest
+    ((k,v), st) = y
+    dest2 = empty(dest, typeof(k), typeof(v))
+    dest2[k] = v
+    grow_to!(dest2, itr, st)
+end
+
+# this is a special case due to (1) allowing both Pairs and Tuples as elements,
+# and (2) Pair being invariant. a bit annoying.
+function grow_to!(dest::AbstractDict{K,V}, itr, st) where V where K
+    y = iterate(itr, st)
+    while y !== nothing
+        (k,v), st = y
+        if isa(k,K) && isa(v,V)
+            dest[k] = v
+        else
+            new = empty(dest, promote_typejoin(K,typeof(k)), promote_typejoin(V,typeof(v)))
+            merge!(new, dest)
+            new[k] = v
+            return grow_to!(new, itr, st)
+        end
+        y = iterate(itr, st)
+    end
+    return dest
+end
+
+empty(a::AbstractDict, ::Type{K}, ::Type{V}) where {K, V} = SwissDict{K, V}()
+
+#hashindex(key, sz) = (((hash(key)%Int) & (sz-1)) + 1)::Int
+
+##Simd utilities
```

Comment:
```suggestion
##SIMD utilities
```

Also fun, i haven't touched these `llvmcall` for SIMD in ages.